### PR TITLE
get addresses from location

### PR DESCRIPTION
**Pull request by ha2san**

### CODE REVIEW

File: app/src/androidTest/java/com/github/brugapp/brug/ItemInformationActivityTest.kt [hunk 1]
```diff
@@ -61,8 +63,9 @@ class ItemInformationActivityTest {
 
     @Test
     fun noLocationAndOwnerAndDateYet() {
-        onView(withId(R.id.item_last_location)).check(matches(withText(str)))
-        onView(withId(R.id.item_owner)).check(matches(withText(str)))
+        val ouchy = "Av. Emile-Henri-Jaques-Dalcroze 7, 1007 Lausanne, Switzerland"
+        onView(withId(R.id.item_last_location)).check(matches(withText(ouchy)))
+        //onView(withId(R.id.item_last_location)).perform(click())
```
Comment (hamzahass) [inline app/src/androidTest/java/com/github/brugapp/brug/ItemInformationActivityTest.kt:68]: Do we still need this comment ?

**Review by hamzahass — APPROVED** [review]
Good work! This was one of our last features and it is nice to have. Consider my two comments below.

File: app/src/androidTest/java/com/github/brugapp/brug/sign_in/SignInActivityButtonTest.kt [hunk 2]
```diff
@@ -46,25 +46,25 @@ class SignInActivityButtonTest {
...
-        runBlocking {
-            firebaseAuth.createUserWithEmailAndPassword(email,password).await()
-        }
+    //@Test
+    //fun demoButtonGoesToItemsMenuActivity() {
+    //    val email = "unlost.app@gmail.com"
+    //    val password = "brugsdpProject1"
+    //    runBlocking {
+    //        firebaseAuth.createUserWithEmailAndPassword(email,password).await()
+    //    }
```
[the remark this answers is not in the document]
Comment (ha2san) [reply app/src/androidTest/java/com/github/brugapp/brug/sign_in/SignInActivityButtonTest.kt:55]: ok thanks

File: app/src/androidTest/java/com/github/brugapp/brug/ItemInformationActivityTest.kt (hunk 1 above)
Comment (ha2san) [reply app/src/androidTest/java/com/github/brugapp/brug/ItemInformationActivityTest.kt:68]: Oh I had forgotten to delete it

File: app/src/main/java/com/github/brugapp/brug/view_model/ItemInformationViewModel.kt [hunk 3]
```diff
@@ -1,26 +1,56 @@
...
+            )
+            if (addresses.isNotEmpty()) {
+                addresses[0].getAddressLine(0)
+            } else {
+                Log.d("ERROR", "addresses empty:")
+                "(${lastLocation.getLatitude()}, ${lastLocation.getLongitude()})"
+            }
+        } catch (e: Exception) {
+            Log.d("ERROR", "Exception detected")
+            return "Not available"
```
Comment (omarelmalki) [inline app/src/main/java/com/github/brugapp/brug/view_model/ItemInformationViewModel.kt:47]: Does "Not available" mean that there was an error in the process and "Not set" mean no location was set ?

**Review by omarelmalki — APPROVED** [review]
Great job on adding the last location in the item menu ! See my question below.

File: app/src/main/java/com/github/brugapp/brug/view_model/ItemInformationViewModel.kt (hunk 3 above)
Comment (ha2san) [reply app/src/main/java/com/github/brugapp/brug/view_model/ItemInformationViewModel.kt:47]: Exactly

**Pull request merged**